salt/client/__init__.py

```diff
@@ -567,10 +567,14 @@ def cmd_batch(
         if 'gather_job_timeout' in kwargs:
             opts['gather_job_timeout'] = kwargs['gather_job_timeout']
 
+        eauth=kwargs.get('eauth', {})
```

Comment:
Almost. You'll need `kwargs['eauth']` too.

One other thought: those four values should be pop'ed out of kwargs so that we don't accidentally send them to minions.